### PR TITLE
8 various fixes

### PR DESCRIPTION
**Pull request by alexasselin008**
docs: #24 add root:true to documentation
docs: #23 add .eslintignore documentation
docs: #22 fix monorepo tsconfig doc
feat: #21 All typescript-configs should have allowImportingTsExtensio… 
feat: #20 packages tsconfig files shouldn't include compilerOptions and exclude fields
feat: #19 clean, test and lint:types scripts shouldn't be delegated t… 
chore: #18 clean script != reset script 
feat: #17 tsup setup

### CODE REVIEW

File: packages/eslint-plugin/README.md [hunk 1]
```diff
@@ -67,10 +70,25 @@ To use this configuration, add the following to your `.eslintrc.json` file:
 ```json
 {
     "$schema": "https://json.schemastore.org/eslintrc",
+    "root": true,
     "extends": "plugin:@workleap/typescript-library"
 }
 ```
 
+## Eslint ignore
```
Comment (patricklafrance) [inline packages/eslint-plugin/README.md:78]: Picky comment but usually it written "ESLint", not "Eslint"

File: packages/typescript-configs/web-application.json [hunk 2]
```diff
@@ -2,7 +2,6 @@
     "$schema": "https://json.schemastore.org/tsconfig",
     "extends": "./react.json",
     "compilerOptions": {
-        "allowImportingTsExtensions": true,
         "moduleResolution": "bundler"
```
Comment (patricklafrance) [inline packages/typescript-configs/web-application.json:5]: I there a specific reason why we haven't add the default `exclude` to this config?

File: packages/eslint-plugin/README.md [hunk 3]
```diff
@@ -67,10 +70,25 @@ To use this configuration, add the following to your `.eslintrc.json` file:
...
+
+You can configure ESLint to ignore certain files and directories while linting by specifying one or more glob patterns.
+Heres the eslint ignore file we usually use at Workleap:
```
Comment (patricklafrance) [inline packages/eslint-plugin/README.md:81]: eslint -> ESLint

**Comment by patricklafrance** [comment]
For the typescript docs migration example, you should probably add `moduleResolution` to `bundler`.

```json
{
  "extends": "@workleap/typescript-configs/web-application", // or any other configurations from this package
  "compilerOptions": {
    "module": "commonjs",
    "moduleResolution": "bundler"
  }
}
```

File: tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,9 @@
+{
+    "extends": "@workleap/typescript-configs/monorepo-root.json",
+    "compilerOptions": {
+        "module": "CommonJS",
+        "outDir": "dist"
+    },
+    "include": ["lib", "tests"],
```
Comment (patricklafrance) [inline tsconfig.json:7]: Also not sure you need an `include` here.

File: tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+    "exclude": ["dist"]
```
Comment (patricklafrance) [inline tsconfig.json:8]: You added `node_modules` to every other excludes, do you also need it here?

File: packages/typescript-configs/web-application.json (hunk 2 above)
Comment (alexasselin008) [reply packages/typescript-configs/web-application.json:5]: I'm not sure what we would exclude here in a web application. There is no dist folders

**Review by patricklafrance — APPROVED** [review]

**Pull request merged**